dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/JdbcEventStore.java

```diff
@@ -1375,6 +1375,13 @@ private StringBuilder dataElementAndFiltersSql(
           .append(" as ")
           .append(deUid);
 
+      if (filterContainsOnlyUnaryOperator(filters)) {
```

Comment:
Why do we need to treat only unary differently from cases below?

Reply:
This one I doubted a little bit.
Do we care about the DE being an option set when the filter contains a unary operator?
If all we want to do is check whether something exists, shouldn't we just check the JSONB in events and forget about the `optionvalue` join?

Reply:
@enricocolasante I think I need your help here 😅 🤔 

@muilpp are you saying that this code is an optimization? So if all a user cares about is if an event has a data value for a DE than we skip the join to `optionvalue`.

Reply:
when filtering a DE that has an option set, a join like the one below is added to the query:
```
	inner join optionvalue as opt_1 on
		lower(opt_1.code) = lower(ev.eventdatavalues #>> '{AAaJGnWR5js, value}')
		and opt_1.optionsetid = :optset_1
		and lower(opt_1.name) = :parameter_2
```
`opt_1` is not used anywhere else, so it seems to me this is just to validate the value in the event DE exists in the corresponding `optionvalue` entry, which sounds like something we should do while importing, isn't it?

that said, what happens when a user deletes a certain option value? is that value deleted from events too? if it's not, then this join makes sense.